include/mbedtls/timing.h

```diff
@@ -35,11 +35,20 @@ extern "C" {
 // Regular implementation
 //
 
+// This macro is for Mbed TLS internal use only
+#if defined(_WIN32) && !defined(EFIX64) && !defined(EFI32)
+#include <windows.h>
+#define MBEDTLS_TIMING_HR_TIME_TYPE_MACRO LARGE_INTEGER
+#else
+#include <sys/time.h>
+#define MBEDTLS_TIMING_HR_TIME_TYPE_MACRO struct timeval
+#endif
+
 /**
  * \brief          timer structure
  */
 struct mbedtls_timing_hr_time {
-    unsigned char MBEDTLS_PRIVATE(opaque)[32];
+    MBEDTLS_TIMING_HR_TIME_TYPE_MACRO MBEDTLS_PRIVATE(opaque);
```

Comment:
why not just do

```
uint64_t MBEDTLS_PRIVATE(opaque)[4];
```

Reply:
It seems more brittle (because we're making an assumption about the size `struct timeval` or equivalent), and wastes some memory. Better to use the actual type and guarantee that it's the correct size and alignment?

Reply:
Yes, but this is actually just a placeholder structure of up to 32 bytes that implementations can use how they want.No point making it specific for one of the implementations - better to have the placeholder structure force the correct alignment (and I think 64 bits should be "enough for anyone" 😊)

Reply:
By "implementation" I meant "choice of implementation in the `.c` file" - i.e. there's currently Windows and POSIX, but Mbed TLS-supported alternatives could currently be added in the `.c` file without needing to update the `.h` file.

If you want to go this way then you might just as well get rid of the currently-private-to-the-.c-file `struct _hr_time`, which would simplify the whole thing